sale_stock_line_customer_ref/models/stock_move.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    sale_line_with_customer_ref_id = fields.Many2one(
```

Comment:
Done, and:
- rewrite the recursive function to not be recursive anymore using a while loop.
- add a `pre_init_hook` script to avoid the recomputation of all moves (useless, and avoid a MemoryError on big DB)